java-client/src/main/java/org/opensearch/client/json/jackson/JacksonJsonpMapper.java

```diff
@@ -54,18 +54,20 @@ public class JacksonJsonpMapper extends JsonpMapperBase {
     private final ObjectMapper objectMapper;
 
     public JacksonJsonpMapper(ObjectMapper objectMapper) {
-        this(objectMapper, new JsonFactory());
+        // Creating the json factory from the mapper ensures it will be returned by JsonParser.getCodec()
+        this(objectMapper, objectMapper.getFactory());
     }
 
     public JacksonJsonpMapper(ObjectMapper objectMapper, JsonFactory jsonFactory) {
         this.provider = new JacksonJsonProvider(jsonFactory);
-        this.objectMapper = objectMapper
-            .configure(SerializationFeature.INDENT_OUTPUT, false)
-            .setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        this.objectMapper = objectMapper;
     }
 
     public JacksonJsonpMapper() {
-        this(new ObjectMapper());
+        this(new ObjectMapper()
+            .configure(SerializationFeature.INDENT_OUTPUT, false)
```

Comment:
Curious, is this change related to the backport?

Reply:
More or less (https://github.com/elastic/elasticsearch-java/blob/main/java-client/src/main/java/co/elastic/clients/json/jackson/JacksonJsonpMapper.java#L56) when back porting the Jackson buffered lookahead change, I also saw that the constructor was different, and saw that they ES improved the constructor so if we pass an existing ObjectMapper it doesn't changes its configuration. For me it is a win, since we use a singleton instance of objectmapper and we had to clone it to construct the opensearch client... since it sets `JsonInclude.Include.NON_NULL` and we use `JsonInclude.Include.NON_ABSENT`... so this change actually makes the constructor with the objectmapper param not change the objectmapper config as it was doing before.